sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,67 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-type birthdays() :: [#{m := 1..12, d := 1..31}].
+
+-reflect_type([ip/0, birthdays/0]).
+
+-export([namespaces/0, fields/1]).
+
+namespaces() -> ["foo", "a.b"].
+
+fields("foo") ->
+    [ {"setting", fun setting/1}
+    , {"endpoint", fun endpoint/1}
+    , {"greet", fun greet/1}
+    , {"numbers", fun numbers/1}
```

Comment:
make sense to reference `"a.b"` here?
e.g. add another field`, {"ab", fun ab/1}`

```
ab(mapping) -> undefined; % so it's just a type checker for `foo.ab`
ab(type) -> fields("ab").
% if fields have logical connections, here is the place to validate
ab(validator) -> fun(V) -> maps:get(<<"f1">>, V) > maps:get(<<"f2">>, V) end;
ab(_) -> undefined.
```

Reply:
I'm not sure how `ab(type) -> typerefl:map()` should work.

Reply:
then there is also a need for

`ab(type) -> array(fields("ab"))`

Reply:
what do you mean by `ab(type) -> fields("ab").` ?

Reply:
if we consider `fields(Name) -> [{"f1", fun f1/1}]` is to declare type of an 'object',
then `f1(type) -> fields(SubName)` means a nested 'object'

Reply:
`ref` type added.